api/client/src/main/java/org/projectnessie/client/http/NessieApiCompatibilityFilter.java

```diff
@@ -28,22 +28,24 @@ public class NessieApiCompatibilityFilter implements RequestFilter {
   private static final String MAX_API_VERSION = "maxSupportedApiVersion";
   private static final String ACTUAL_API_VERSION = "actualApiVersion";
 
-  private HttpClient.Builder builder;
   private final int clientApiVersion;
   private final AtomicBoolean checkDone = new AtomicBoolean(false);
 
-  NessieApiCompatibilityFilter(HttpClient.Builder builder, int clientApiVersion) {
-    this.builder = builder.copy().clearRequestFilters().clearResponseFilters();
+  private volatile HttpClient httpClient;
```

Comment:
Does it really need to be `volatile`? It appears to be set by the same thread that creates this object :thinking: